### PR TITLE
Harmonize statistics pipelines

### PR DESCRIPTION
**Pull request by alexandreroutier**
Content of this PR:

## Refactor `statistics-surface`
- [X] `build_input_node`
     - [X] Remove current read_parameters_node	
     - [X] With the help of `statistics_surface_utils.py::check_inputs`, use new `clinica_file_reader` function to check and extract surface-based features	
     - [X] Delete `statistics_surface_utils.py::check_inputs function`
     - [X] Move `statistics_surface_cli.py` checks of input data in this method	
     - [X] Display participants, covariates and info regarding the GLM.	
- [X] `build_core_nodes`	
     - [X] Use `surfstat.inputs.full_width_at_half_maximum = self.parameter['full_width_at_half_maximum']`, instead of connecting `read_parameters_node.inputs.full_width_at_half_maximum` to `surfstat.inputs.full_width_at_half_maximum`	
     - [X] Repeat for other keys	
     - [X] Use working directory	
- [X] build_output_node	
     - [X] Remove `path_to_matscript` and `freesurfer_home`: it should be set in `runmatlab` function	
     - [X] Copy results from `<WD>` to `<CAPS>` (outputs are currently generated into CAPS folder)
- [X] Clean/adapt `statistics_surface_utils.py`	
- [X] Add FreeSurfer to info.json

(Handle overwrite case is left for later)

- [X] Remove `--threshold_[un]corrected_pvalue` flags`
- [X] Remove str_format argument

## Harmonize `statistics-surface` and `statistics-volume` 
- [X] Harmonize measure label and grabbing of input data
- [X] Use group label wording instead of group ID
- [X] Create CLI category 'Pipeline options if you use inputs from pet-{surface|volume} pipeline:'
- [X] Create CLI category 'Pipeline options if you use inputs from your custom pipeline:' 

Harmonization of how PET tracers are handles will be done in another PR since it will impact `pet-volume`,  `pet-surface` and `machinelearning-prepare-spatial-svm`. pipeline 

## Misc
- [X] Add have_same_subjects function

### CODE REVIEW

**Comment by alexandreroutier** [comment]
Hi @mdiazmel,

This refactoring of `statistics-surface` pipeline is now ready for review! :)

My text editor automatically removed spaces in `clinica/lib/clinicasurfstat/clinicasurfstat.m` but the only lines to look at this file are:
```matlab
            cell_factor = char(term(tsvdata{indexunique}));
            factor1 = cell_factor{1};
            factor2 = cell_factor{2};
```
These lines solve the "Indexing with parentheses" error mentioned in Issue #90 🤞 

I will do another PR to handle covariates for `statistics-volume` pipelines and another one for PET tracers involved in several Clinica pipelines.

Alex

File: clinica/pipelines/statistics_surface/statistics_surface_utils.py [hunk 1]
```diff
@@ -27,246 +27,271 @@ def get_fdg_pet_surface_custom_file():
...
-    # matlab.input_spec.add_trait(matlab.input_spec(), 'nosoftwareopengl', openGL_trait() )
+    # Add the dynamic traits
+    # opengl_trait = traits.Bool(
+    #     True, argstr='-nosoftwareopengl', usedefault=True, desc='Switch on hardware openGL', nohash=True
+    # )
+    # matlab.input_spec.add_trait(matlab.input_spec(), 'nosoftwareopengl', opengl_trait())
     if sys.platform.startswith('linux'):
-        matlab.inputs.args = '-nosoftwareopengl'  # Bug, for my laptop, it does not work, but the command line does have the flag -nosoftwareopengl, we should try on other computer's matlab to check if this flag works!
-    matlab.inputs.paths = path_to_matscript  # CLINICA_HOME, this is the path to add into matlab, addpath
+        # Bug(JW): for my laptop, it does not work, but the command line does have the flag -nosoftwareopengl,
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/pipelines/statistics_surface/statistics_surface_utils.py:192]: Junhao Wen aka @anbai106 :)

File: clinica/pipelines/statistics_surface/statistics_surface_utils.py [hunk 2]
```diff
@@ -27,246 +27,271 @@ def get_fdg_pet_surface_custom_file():
...
+        # TODO: We should try on other computer's matlab to check if this flag works!
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/pipelines/statistics_surface/statistics_surface_utils.py:193]: > This has been done? Normally, the right way to launch Matlab from the terminal includes `-nodesktop -nosoftwareopengl`

This is a very old message that I did not delete so I can not say about it. But I can test on Mac & Linux to see if anything goes well or wrong now.

**Comment by alexandreroutier** [comment]
Hi @mdiazmel,

Thanks for the feedback. Based on your comments and your question regarding Matlab windows, I will have a look into this question. I know that when running `statistics-surface` or  `statistics-volume` pipelines, some Matlab windows pop up but I don't remember if this could be removed 🤔 

So I will test a few things before closing this question.

Alex

File: clinica/pipelines/statistics_surface/statistics_surface_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/pipelines/statistics_surface/statistics_surface_utils.py]: According to [Nipype's documentation](https://nipype.readthedocs.io/en/latest/api/generated/nipype.interfaces.matlab.html), the `-nodesktop` is activated on default. 

Moreover, from [Matlab documentation](https://fr.mathworks.com/help/matlab/ref/opengl.html):
> hardware — Hardware-accelerated OpenGL
hardware
Hardware-accelerated OpenGL, specified as hardware. All systems support using the opengl hardware command to switch from basic hardware to hardware OpenGL. However, only Windows systems support using the opengl hardware command to switch from software to hardware OpenGL. To switch from software to hardware on Linux systems, start MATLAB with the -nosoftwareopengl flag. If your system automatically switched to using software OpenGL, then forcing your system to use hardware OpenGL can cause instabilities.

Based on these two points, I have removed the `-nosoftwareopengl` flag (bfca8c7c) and non regression tests passed on Mac (my laptop) and Linux ("Aramis 3").

**Comment by mdiazmel** [comment]
I'll merge this one! 
Thanks @alexandreroutier!

**Pull request merged**